server/services/transactionsTransformer.go

```diff
@@ -191,11 +195,48 @@ func (transformer *transactionsTransformer) moveBalanceTxToRosetta(tx *transacti
 		Amount:  transformer.extension.valueToNativeAmount("-" + tx.InitiallyPaidFee),
 	})
 
+	innerTxOperationsIfRelayedCompletelyIntrashardWithSignalError, err := transformer.extractInnerTxOperationsIfRelayedCompletelyIntrashardWithSignalError(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	operations = append(operations, innerTxOperationsIfRelayedCompletelyIntrashardWithSignalError...)
+
 	return &types.Transaction{
 		TransactionIdentifier: hashToTransactionIdentifier(tx.Hash),
 		Operations:            operations,
 		Metadata:              extractTransactionMetadata(tx),
+	}, nil
+}
+
+func (transformer *transactionsTransformer) extractInnerTxOperationsIfRelayedCompletelyIntrashardWithSignalError(tx *transaction.ApiTransactionResult) ([]*types.Operation, error) {
+	isRelayedTransaction := isRelayedV1Transaction(tx)
+	if !isRelayedTransaction {
+		return nil, nil
 	}
+
+	innerTx, err := parseInnerTxOfRelayedV1(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if isZeroBigInt(&innerTx.Value) {
```

Comment:
for token transfers (the other feature branch) should we care also about non native token transfers?
if yes, maybe add a TODO?

Reply:
Good question. Ideally, the existing logic (in `feat/esdt`) that handles logs & events for ESDT transfers should be sufficient. Added an issue here, to check & confirm: https://github.com/multiversx/mx-chain-rosetta/issues/82.